### PR TITLE
Expose membership processing status of dynamic groups

### PR DESCRIPTION
**Pull request by FaithOmbongi**
This PR introduces the `membershipRuleProcessingStatus` parameter to the group resource so customers can track status of dynamic groups processing - a feature previously available only via the Azure administration portal.

### CODE REVIEW

**Comment by yyuank** [comment]
Thanks Faith for adding the examples to list group page. I think get group page (https://docs.microsoft.com/en-us/graph/api/group-get?view=graph-rest-beta&tabs=http) is a better place for "Example 8: Read a specific dynamic group".

**Review by yyuank — APPROVED** [review]
I reviewed changes. Looks good. Thanks so much Faith!

File: api-reference/beta/resources/membershipruleprocessingstatus.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: "membershipRuleProcessingStatus resource type"
+description: "Represents the current status of dynamic group processing."
+localization_priority: Normal
+author: "davidmu1"
```
Comment (davidmu1) [inline api-reference/beta/resources/membershipruleprocessingstatus.md:5]: My alias shouldn't be here.

File: api-reference/beta/resources/membershipruleprocessingstatus.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+| lastMembershipUpdated | edm.DateTime | Most recent date and time when membership of a dynamic group was updated. <br><br> Optional. Read-only.|
+| errorMessage | String | Detailed error message if dynamic group processing ran into an error. <br><br> Optional. Read-only.|
+
+### membershipRuleProcessingStatusDetails values
+
+| Member | Description |
+|:-------- |:----------- |
+| NotStarted | Group has been created or updated, and awaiting processing.|
+| Running | Processing has started.|
+| Succeeded | Processing has completed. Incremental object changes will be processed perpetually. |
```
Comment (davidmu1) [inline api-reference/beta/resources/membershipruleprocessingstatus.md:30]:
```suggestion
| Succeeded | Processing has completed. Incremental object changes are processed perpetually. |
```

File: api-reference/beta/resources/membershipruleprocessingstatus.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+| Failed | Processing ran into an error. See **errorMessage** for details. |
+| UnknownFutureValue | Sentinel value for supporting future unknown values. |
```
Comment (davidmu1) [inline api-reference/beta/resources/membershipruleprocessingstatus.md:32]:
```suggestion
| UnknownFutureValue | Supports future values. |
```

File: concepts/changelog.md [hunk 4]
```diff
@@ -13,6 +13,12 @@ For a summary of the value of these API changes, as well as recent tools, compon
```
Comment (davidmu1) [inline concepts/changelog.md:13]: Don't forget the JSON version.

File: api-reference/beta/resources/membershipruleprocessingstatus.md (hunk 1 above)
Comment (FaithOmbongi) [reply api-reference/beta/resources/membershipruleprocessingstatus.md:5]: Noted. Changing to @yyuank

File: concepts/changelog.md (hunk 4 above)
Comment (FaithOmbongi) [reply concepts/changelog.md]: Resolved.

File: api-reference/beta/api/group-list.md [hunk 5]
```diff
@@ -398,6 +398,71 @@ Content-type: application/json
...
+} -->
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+  "@odata.context": "https://graph.microsoft.com/beta/$metadata#groups(id,membershipRule,membershipRuleProcessingState,membershipRuleProcessingStatus)",
+  "value":[
+    {
+      "id": "7c120147-3f88-447b-8c48-954863301048",
```
Comment (davidmu1) [inline api-reference/beta/api/group-list.md:434]: You only need to show one object in the returned list to give the user the idea of what will be returned.

File: api-reference/beta/resources/group.md [hunk 6]
```diff
@@ -134,6 +134,7 @@ This resource supports:
 |mailNickname|String|The mail alias for the group, unique in the organization. This property must be specified when a group is created. These characters cannot be used in the mailNickName: `@()\[]";:.<>,SPACE`. <br><br>Returned by default. Supports $filter.|
 |membershipRule|String|The rule that determines members for this group if the group is a dynamic group (groupTypes contains `DynamicMembership`). For more information about the syntax of the membership rule, see [Membership Rules syntax](https://azure.microsoft.com/documentation/articles/active-directory-accessmanagement-groups-with-advanced-rules/). <br><br>Returned by default. |
 |membershipRuleProcessingState|String|Indicates whether the dynamic membership processing is on or paused. Possible values are "On" or "Paused". <br><br>Returned by default. |
+|membershipRuleProcessingStatus|[membershipRuleProcessingStatus](membershipruleprocessingstatus.md) |Describes the processing status for rules-based dynamic groups. Property will be `null` for non-rule based dynamic groups or if the dynamic group processing has been paused. <br><br>Returned only on `$select`. Supports `$filter`. Read-only. |
```
Comment (davidmu1) [inline api-reference/beta/resources/group.md:137]:
```suggestion
|membershipRuleProcessingStatus|[membershipRuleProcessingStatus](membershipruleprocessingstatus.md) |Describes the processing status for rules-based dynamic groups. The property is `null` for non-rule based dynamic groups or if the dynamic group processing has been paused. <br><br>Returned only on `$select`. Supports `$filter`. Read-only. |
```

**Review by davidmu1 — APPROVED** [review]

**Pull request merged**